webapp/src/test/java/com/box/l10n/mojito/android/strings/AndroidStringDocumentMapperTest.java

```diff
@@ -782,4 +783,461 @@ private TextUnitDTO textUnitDTO(
+            104L,
+            "name1 _other",
+            "content1_many",
+            "comment1",
+            "my/path0.xml",
+            "many",
+            "name1_other"));
+    textUnits.add(
+        sourceTextUnitDTO(
+            105L,
+            "name1 _other",
+            "content1_other",
+            "comment1",
+            "my/path0.xml",
+            "other",
+            "name1_other"));
+
+    textUnits.add(
+        sourceTextUnitDTO(
+            201L,
+            "name1 _other",
+            "content2_one",
+            "comment1",
+            "my/path0.xml",
+            "one",
+            "name1_other"));
+
+    document = mapper.readFromSourceTextUnits(textUnits);
+
+    assertThat(document).isNotNull();
+    assertThat(document.getSingulars()).hasSize(1);
+    AndroidSingular singular = document.getSingulars().get(0);
+    assertThat(singular.getId()).isEqualTo(123L);
+    assertThat(singular.getName()).isEqualTo("my/path0#@#name0");
+    assertThat(singular.getContent()).isEqualTo("content0");
+    assertThat(singular.getComment()).isEqualTo("comment0");
+
+    assertThat(document.getPlurals()).hasSize(2);
+    AndroidPlural plural = document.getPlurals().getFirst();
+    assertThat(plural.getName()).isEqualTo("my/path0.xml#@#name1");
+    assertThat(plural.getComment()).isEqualTo("comment1");
+    assertThat(plural.getItems()).hasSize(6);
+
+    assertThat(plural.getName()).isEqualTo("my/path0.xml#@#name1");
+    assertThat(plural.getComment()).isEqualTo("comment1");
+    assertThat(plural.getItems()).hasSize(6);
+    assertThat(plural.getItems().get(ZERO).getId()).isEqualTo(100L);
+    assertThat(plural.getItems().get(ZERO).getContent()).isEqualTo("content1_zero");
+    assertThat(plural.getItems().get(ONE).getId()).isEqualTo(101L);
+    assertThat(plural.getItems().get(ONE).getContent()).isEqualTo("content1_one");
+    assertThat(plural.getItems().get(TWO).getId()).isEqualTo(102L);
+    assertThat(plural.getItems().get(TWO).getContent()).isEqualTo("content1_two");
+    assertThat(plural.getItems().get(FEW).getId()).isEqualTo(103L);
+    assertThat(plural.getItems().get(FEW).getContent()).isEqualTo("content1_few");
+    assertThat(plural.getItems().get(MANY).getId()).isEqualTo(104L);
+    assertThat(plural.getItems().get(MANY).getContent()).isEqualTo("content1_many");
+    assertThat(plural.getItems().get(OTHER).getId()).isEqualTo(105L);
+    assertThat(plural.getItems().get(OTHER).getContent()).isEqualTo("content1_other");
+
+    plural = document.getPlurals().get(1);
+    assertThat(plural.getName()).isEqualTo("my/path0.xml#@#name1");
+    assertThat(plural.getComment()).isEqualTo("comment1");
+    assertThat(plural.getItems()).hasSize(1);
+    assertThat(plural.getItems().get(ONE).getId()).isEqualTo(201L);
+    assertThat(plural.getItems().get(ONE).getContent()).isEqualTo("content2_one");
+  }
+
+  @Test
+  public void testReadFromSourceTextUnitsWithDuplicatedOtherPluralsWithDifferentContent() {
```

Comment:
This is the test for when the other plural source is modified:
`<!-- %1$d: number of sections we have in the board -->
    <plurals name="plural_homefeed_section_count">
        <item quantity="one">\u0020· %1$d section</item>
        <item quantity="other">\u0020· %1$d modified sections</item>
    </plurals>`